### ISSUE TITLE
enabling SQL DB to *pull* changes form the repo for updating a project

### ISSUE THREAD

**Issue by BlackFoundry** [BlackFoundryCom/robo-cjk#16]
or similar mechanism for taking into account changes made outside the SQL API.
A bit similar to the initial 'import' of ZIP .rcjk project but more flexible

**Comment by fabiocaccamo** [comment]
We said that once the project were migrated to MySQL, the database would have kept the data centralized instead of git. Now this request goes in the opposite direction, why we would need this now? 
Adding this feature would be equivalent to the initial import, and as the initial import it will be slow.

**Comment by BlackFoundry** [comment]
I refer @justvanrossum note: "[git] should not be for backup, it should be the main data storage. mysql should be a speedy caching frontend that makes it better for users"
I am open to suggestions but I tend to agree with him here. We need to be able (if only for bug fixing) to act on the git repo as well, and that these changes are reflected in the frontend.

**Comment by justvanrossum** [comment]
I don't see how this should be as slow as an initial import: it only needs to read the files that have been changed.

I find it highly undesirable for the mysql frontend to be the _only_ way to modify a project.

**Comment by fabiocaccamo** [comment]
if you want to go fast it doesn't have to be the slowest (git) to lead :)

**Comment by justvanrossum** [comment]
The pull doesn't need to be instantaneous, it could be timed like the push. Although a force pull/push would be important to have, too.

**Comment by fabiocaccamo** [comment]
But if it is now instantaneous, someone could do some changes in the meantime and this will generate conflicts.

**Comment by justvanrossum** [comment]
That is true, and that's a good reason for this to be an very exceptional case. Someone who is not mysql should manually ensure that it is safe to edit. 

Maybe it is acceptable that if mysql detects a conflict, it simply commits the conflict with conflict markers and all, to be fixed manually.

Let me think. Perhaps there's a better way.
